### DISCUSSION TITLE
how to use scuttle to make sure itsio side cards are killed

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18485] [Q&A]
Hey everyone, I have a question about istio behaviors with k8s and dagster that I figured would be best directed your way. Specifically, I want to add scuttle to my dagster deployment in k8s in order to make sure that the istio sidecars are killed once main dagster job/step containers are done running for pods. How should I go about doing this? Are there any recommended ways to add `scuttle`

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C014N0PK37E/p1701725942726379?thread_ts=1701725942.726379&cid=C014N0PK37E).*

**Comment by ian-scale** [comment]
This one had an easy solution - you can just add scuttle to your dockerfile and set an entrypoint. Like this: 

```
COPY --from=kvij/scuttle:1.0.4 /scuttle /bin/scuttle
ENV ENVOY_ADMIN_API=...
ENV START_WITHOUT_ENVOY=...
ENV ISTIO_QUIT_API=...
...
ENTRYPOINT ["scuttle"]
```

**Reply by OrenLederman** [reply]
We use this operator - https://github.com/otherguy/k8s-controller-sidecars

You can define which containers are sidecars, and it'll kill them when the main container is done. This way, you know need to modify anything in the main container.

Also - sidecars are supposed to get better support in k8s 1.28 🤞
